server/services/applicationService.ts

```diff
@@ -80,6 +87,37 @@ export default class ApplicationService {
     }
   }
 
+  private getResponsesForTask(application: Application, taskName: string): Array<PageResponse> {
+    const pageNames = Object.keys(application.data[taskName])
+    const responsesForPages = pageNames.map(pageName => this.getResponseForPage(application, taskName, pageName))
+    return responsesForPages
+  }
+
+  private getResponseForPage(application: Application, taskName: string, pageName: string): PageResponse {
+    const Page = this.getPage(taskName, pageName)
+
+    const body = application?.data?.[taskName]?.[pageName]
+    const page = new Page(body, application)
+
+    return page.response()
+  }
+
+  private getPage(taskName: string, pageName: string) {
+    const pageList = pages[taskName]
+
+    const Page = pageList[pageName]
+
+    if (!Page) {
+      throw new UnknownPageError()
+    }
+
+    return Page
+  }
+
+  private firstPageForTask(taskName: string) {
+    return Object.keys(pages[taskName])[0]
```

Comment:
I was ready to say we shouldn't be relying on the order of properties in an object but [looks like I'm wrong](https://stackoverflow.com/questions/30076219/does-es6-introduce-a-well-defined-order-of-enumeration-for-object-properties)

Reply:
I'm not even convinced we actually need this now, but it's assumed in a lot of tests, so thought I'd keep it in for the sake of simplicity